src/features/comments/commentsSlice.ts

```diff
@@ -0,0 +1,106 @@
+/* eslint-disable no-param-reassign */
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import {
+  createComment,
+  deleteComment,
+  getPostComments,
+} from '../../api/comments';
+import { Comment } from '../../types/Comment';
+
+type CommentsType = {
+  items: Comment[];
+  loaded: boolean;
+  hasError: boolean;
+};
+
+const initialState: CommentsType = {
+  items: [],
+  loaded: false,
+  hasError: false,
+};
+
+// const loadComments = createAsyncThunk(
+//   'comments/fetchComments',
+//   async (postId: number) => {
+//     await getPostComments(postId);
+//   },
+// );
+
+export const loadComments = createAsyncThunk(
+  'comments/fetchComment',
+  async (postId: number) => {
+    const value = await getPostComments(postId);
+
+    return value;
+  },
+);
+
+// const addComment = createAsyncThunk(
+//   'comments/addComment',
+//   async (comment: Omit<Comment, 'id'>) => {
+//     createComment(comment);
+//   },
+// );
+
+export const addComment = createAsyncThunk(
+  'comments/addComment',
+  async (comment: Omit<Comment, 'id'>) => {
+    const value = await createComment(comment);
+
+    return value;
+  },
+);
+
+// const delateComment = createAsyncThunk(
+//   'comments/delateComments',
+//   async (commentId: number) => {
+//     deleteComment(commentId);
+//   },
+// );
+
+export const delateComment = createAsyncThunk(
+  'comments/deleteComment',
```

Comment:
There is a typo in the action name `delateComment`. It should be `deleteComment` to match the intended functionality and avoid errors.